pkg/types/plugin_service.go

```diff
@@ -0,0 +1,28 @@
+package types
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+)
+
+type PluginServiceConfig struct {
+	ProviderType string
+	Command      string
+	PluginName   string
+	PluginConfig []byte
+}
+
+// PluginClient is the client interface to a plugin running
+// as a generic job (job type = GenericPlugin) inside the core node.
+type PluginClient interface {
+	NewPluginServiceFactory(ctx context.Context, config PluginServiceConfig, grpcProvider grpc.ClientConnInterface, errorLog ErrorLog) (ReportingPluginFactory, error)
```

Comment:
Would it make sense to align with the type name here?
```suggestion
	NewReportingPluginFactory(ctx context.Context, config PluginServiceConfig, grpcProvider grpc.ClientConnInterface, errorLog ErrorLog) (ReportingPluginFactory, error)
```
Or even more broadly with other types too?